src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs

```diff
@@ -739,8 +739,9 @@ private static IEnumerator<T> GetEnumerator(Node? head)
             Node? current = head;
             while (current != null)
             {
-                yield return current._value;
+                T value = current._value;
                 current = current._next;
+                yield return value;
             }
```

Comment:
@jjonescz, without this change, the test was consistently failing:

```sh
$ ./build.sh libs+libs.tests
$ ./dotnet.sh build -t:Test src/libraries/System.Collections.Concurrent/tests
...
  Starting:    System.Collections.Concurrent.Tests (parallel test collections = on [4 threads], stop on fail = off)
    System.Collections.Concurrent.Tests.ConcurrentStackTests.Enumerator_MoveNext_AfterDisposal(count: 1) [FAIL]
      System.NullReferenceException : Object reference not set to an instance of an object.
      Stack Trace:
        /_/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs(743,0): at System.Collections.Concurrent.ConcurrentStack`1.GetEnumerator(Node head)+MoveNext()
        /_/src/libraries/Common/tests/System/Collections/IEnumerable.Generic.Tests.cs(395,0): at System.Collections.Tests.IEnumerable_Generic_Tests`1.Enumerator_MoveNext_AfterDisposal(Int32 count)
           at InvokeStub_IEnumerable_Generic_Tests`1.Enumerator_MoveNext_AfterDisposal(Object, Span`1)
        /_/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodBaseInvoker.cs(95,0): at System.Reflection.MethodBaseInvoker.InvokeWithOneArg(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
    System.Collections.Concurrent.Tests.ConcurrentStackTests.Enumerator_MoveNext_AfterDisposal(count: 75) [FAIL]
...
```
is it expected?